### PR TITLE
Image size typedef

### PR DESCRIPTION
**Pull request by adamrankin**
Well, this one hits a lot of files.

I did my best to find all uses of framesize, imagesize, etc...

For devices that don't require external SDKs, I think I've got everything. For devices that require SDKs, I did a search and replace approach, so there may be compiler errors/warnings.

### CODE REVIEW

**Comment by dzenanz** [comment]
With a move to C++11, this is a logical and welcome change!

**Comment by adamrankin** [comment]
A review would be most welcome. I'm sure I've missed some parts, and there were some changes in the Capistrano class.

File: src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx [hunk 1]
```diff
@@ -683,15 +685,15 @@ PlusStatus vtkPlusCapistranoVideoSource::WriteConfiguration(vtkXMLDataElement* r
...
     LOG_WARNING("vtkPlusIntersonVideoSource is expecting one output channel and there are "
-                 << this->OutputChannels.size() << " channels. First output channel will be used.");
+                << this->OutputChannels.size() << " channels. First output channel will be used.");
   }
 
-  if( this->OutputChannels.empty() )
+  if (this->OutputChannels.empty())
   {
-    LOG_ERROR("No output channels defined for vtkPlusIntersonVideoSource. Cannot proceed." );
+    LOG_ERROR("No output channels defined for vtkPlusIntersonVideoSource. Cannot proceed.");
```
Comment (dzenanz) [inline src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx]: This should be vtkPlusCapistranoVideoSource. I guess Capistrano wrapper was based on Interson one :smile:

File: src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx [hunk 2]
```diff
@@ -866,24 +868,23 @@ PlusStatus vtkPlusCapistranoVideoSource::SetupProbe(int probeID)
...
   if (display == NULL)
   {
     LOG_ERROR("Could not initialize the display");
     return PLUS_FAIL;
   }
 
   // Initialize DIB -----------------------------------------------------------
-  int frameSize[2] = {imageSize[0], imageSize[1]};
-  this->Internal->InitializeDIB(frameSize);
+  this->Internal->InitializeDIB(imageSize);
```
Comment (dzenanz) [inline src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx]: This seems to be the most significant change.

File: src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx [hunk 3]
```diff
@@ -1589,20 +1589,20 @@ PlusStatus vtkPlusCapistranoVideoSource::SetDepthMm(float depthMm)
...
 }
 
 // ----------------------------------------------------------------------------
 PlusStatus vtkPlusCapistranoVideoSource::UpdateUSProbeParameters()
 {
   // Set US Probe scan mode -------------------------------------------------
-  if(this->BidirectionalMode)
-    usbSetBidirectionalMode();
+  if (this->BidirectionalMode)
+  { usbSetBidirectionalMode(); }
```
Comment (dzenanz) [inline src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx]: Doesn't the style call for braces on lines by themselves to emphasize an if? This is hardly better, if not even worse than no braces.

File: src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.h [hunk 4]
```diff
@@ -270,8 +269,8 @@ class vtkPlusDataCollectionExport vtkPlusCapistranoVideoSource: public vtkPlusUs
...
 
 private:
-  vtkPlusCapistranoVideoSource(const vtkPlusCapistranoVideoSource &); // Not implemented
-  void operator=(const vtkPlusCapistranoVideoSource &); // Not implemented
+  vtkPlusCapistranoVideoSource(const vtkPlusCapistranoVideoSource&);  // Not implemented
+  void operator=(const vtkPlusCapistranoVideoSource&);  // Not implemented
 };
 
-#endif
+#endif
```
Comment (dzenanz) [inline src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.h]: Perhaps add a newline at the end?

**Comment by dzenanz** [comment]
@jperdomo23 @jamesobutler Can you test this PR?

File: src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx (hunk 3 above)
Comment (adamrankin) [reply src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx]: Yes, my astyle formatter was not set up correctly. It should place them in brackets on their own lines. I will correct.

File: src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.h (hunk 4 above)
Comment (dzenanz) [reply src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.h]: https://stackoverflow.com/a/37684763/276168

Comment (adamrankin) [reply src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.h]: Aaaahhh... I will stop removing them!

**Comment by jamesobutler** [comment]
Sure, I can test the capistrano class once changes are complete.

**Comment by adamrankin** [comment]
@lassoan @cpinter I'd like to merge this (I am testing my current master merge now), before more commits make it harder to merge.

I've done my best, but suspect I'll still have broken some things. I have time to address them, so is now an ok time to merge (after merging in master to this branch)?

**Comment by lassoan** [comment]
As I wrote above, I'm OK with merging this.

**Comment by adamrankin** [comment]
Ok, running tests now and then will merge.

**Comment by cpinter** [comment]
Looks good to me, thanks! What is the main reason for switching to std::array?

**Comment by adamrankin** [comment]
Personal preference. I wanted a std type so that all of the std algorithms
could work on it if necessary.

I also wanted a fixed size of 3, to reduce the amount of if(size == 2)
etc... Only the devices need set the third element to 1 and never touch it
again (for 2d devices).

**Comment by cpinter** [comment]
@adamrankin Yes the fixed size of 3 makes complete sense to me. I was curious about all the other variables you changed. Thanks!

**Comment by adamrankin** [comment]
I guess I tend to prefer types like std::array<int, 3> over int* or int[3].

They seem less prone to ambiguity or misintepretation. I know good variable naming will help identify the meaning of a *, but std::array<int, 3> is always clear (in my opinion).

**Comment by adamrankin** [comment]
Plus, std::array<int, 3> as a return type will be returned by copy, meaning less error checking on return values.

**Pull request merged**